cvmfs/util.cc

```diff
@@ -1373,6 +1373,64 @@ bool ManagedExec(const vector<string>  &command_line,
   return true;
 }
 
+
+/**
+ * Traverses the $PATH environment variable to find the absolute path of a given
+ * program name. Pretty much what `which` in bash would do.
+ *
+ * Inspired by a solution to this bulletin board question:
+ *  http://www.linuxquestions.org/questions/programming-9/g
+ *         et-full-path-of-a-command-in-c-117965/#post611028
+ *
+ * @param exe_name  the name of the program to search the $PATH for
+ * @return          absolute path to the program or empty string if not found
+ */
+std::string GetExecutablePath(const std::string &exe_name) {
+  std::string result;
+
+  if (exe_name.empty() || exe_name.find('/') != std::string::npos) {
+    return result;
+  }
+
+  const char *searchpath = getenv("PATH");
+  if (NULL == searchpath) {
+    return result;
+  }
+
+  const std::vector<std::string> paths = SplitString(searchpath, ':');
+        std::vector<std::string>::const_iterator i    = paths.begin();
+  const std::vector<std::string>::const_iterator iend = paths.end();
+  for (; i != iend; ++i) {
+    const std::string candidate_path = *i + "/" + exe_name;
+
+    char *real_path = realpath(candidate_path.c_str(), NULL);
+    if (NULL == real_path) {
+      continue;
+    }
+
+    struct stat statinfo;
+    const int res = stat(real_path, &statinfo);
```

Comment:
`stat()` in cvmfs is handled by the portable (64bit libc interface on Linux and OS X) `platform_stat`